### PR TITLE
turso-cli: 0.85.3 -> 0.86.3

### PR DESCRIPTION
**Pull request by Fryuni**
## Description of changes

Changes: https://github.com/tursodatabase/turso-cli/compare/v0.85.3...v0.86.3

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kashw2 — APPROVED** [review]
LGTM

**Comment by Fryuni** [comment]
All the CLI commands are working, _except_ the login.

I'm talking with the turso team to see if it is something we are doing here. I'll keep this in draft while figuring that out.

**Comment by Fryuni** [comment]
If someone else tried could try it out to see if the login is working then it would mean it is something on my machine, we can merge this in that case.

**Comment by frontsideair** [comment]
I'm on the previous version on nixpkgs (v0.85.3) and login didn't work for me either, but there's `turso auth login --headless` option. I got an environment token as an env var back, but couldn't use anything possibly due to old version. I'll try the version in this PR next.

**Comment by frontsideair** [comment]
I checked out this PR and run `nix run .#turso-cli -- db create` with the token env var set, didn't work. Here's the error:
```sh
Error: failed to get groups: Get "https://api.turso.tech/v1/groups": net/http: invalid header field value for "Tursocliversion"
```

List command didn't work either. Hope this is helpful in some way.

**Comment by Fryuni** [comment]
So, progress on this is on [this thread on their discord](https://canary.discord.com/channels/933071162680958986/1163943641874694214)

TL;DR:

- The problem is really on our derivation.
- I checked out a commit from two months ago and login was working fine (other things were not because old version)
- I bisected to find out when it started breaking, it was on f0a8d6145eb704e46c5bf48edaab719e527bb47c, from my PR #257627. At that point, everything works _except_ logging in, which I forgot to test at the time because I was already logged in.
- Testing each change individually, enabling the `prod` build tag is what makes the login break. AFAICT that build tag is used only to embed the version instead of using a hard-coded `"dev"` value.

---

I have no idea _why_ enabling the `prod` build tag breaks. It does not seem to be just because of embedding since there are other filed that get embedded even on dev build, but then it works fine.

Following the suggestion @kashw2 made from the beginning (https://github.com/NixOS/nixpkgs/pull/257627#discussion_r1339228065) works. I was against it because I though the build tag would be used elsewhere. It isn't, but it could be.

Also, my point [here](https://github.com/NixOS/nixpkgs/pull/257627#discussion_r1340651046) about `-X` ldflag not working for unexported variable is wrong, I don't know why it failed when I tested, probably I did something else incorrectly.

**Comment by Fryuni** [comment]
I figured out why the auth it wasn't working before, it uses the embedded version on the request to the server, but `echo "v${version}" > internal/cmd/version.txt` adds a newline at the end of the file. So the embedded version was `"v0.85.3\n"`.

Turso uses `printf` for that, which doesn't add a newline:
https://github.com/tursodatabase/turso-cli/blob/main/internal/cmd/version_prod.go#L11

**Review by LucioFranco — APPROVED** [review]

**Comment by LucioFranco** [comment]
@lovesegfault you able to take a look?
